metabox/metabox/core/runner.py

```diff
@@ -40,24 +40,33 @@ def __init__(self, args):
         logger.add(
             sys.stdout,
             format=self._formatter,
-            level=args.log_level)
+            level=args.scenario_log_level)
         logger.level("TRACE", color="<w><dim>")
         logger.level("DEBUG", color="<w><dim>")
         # session config
-        if not args.config.exists():
+        if not args.config or not args.config.exists():
             raise SystemExit('Config file not found!')
         else:
             self.config = read_config(args.config)
             self.config = guess_source_uri(self.config)
             validate_config(self.config)
         # effective set of machine configs required by scenarios
         self.combo = set()
-        self.scn_variants = None
+        self.scn_variants = []
         self.machine_provider = None
         self.failed = False
-        self.tags = set(self.args.tags or [])
-        self.exclude_tags = set(self.args.exclude_tags or [])
-        self.hold_on_fail = self.args.hold_on_fail
+        try:
+            self.tags = set(self.args.tags or [])
+        except AttributeError:
+            self.tags = []
```

Comment:
I'm a bit puzzled by this.

- if no tags have been provided, `self.args.tags` will be `None`, so `self.args.tags or []` will return `[]` and `set(self.args.tags or [])` will therefore be an empty set.
- if tags are provided, then `self.args.tags` is a list of strings and therefore `set(self.args.tags or [])` is a set of strings.

I don't see how this could raise an `AttributeError`.

The other thing is that, if this exception occurs anyway, `self.tags` will be a list instead of a set. Probably not a big deal in Python, but still a bit weird to me :)